src/Extension.php

```diff
@@ -41,6 +42,23 @@ public function load(ContainerBuilder $container, array $config): void
         $definition->addArgument($config['fileNamePrefix']);
         $definition->addArgument($config['outputDir']);
 
+        // Integration with other Behat extensions that provide screenshot services.
+        $imageUploaderService = null;
+        if (!empty($config['screenshotExtension'])) {
+          $tags = $container->findTaggedServiceIds('screenshot.service');
+          foreach (array_keys($tags) as $id) {
+            // Check if the container has the definitions for the service.
+            if ($container->hasDefinition($id)) {
+              $service = $container->get($id);
+              // Check if the configuration for the screenshot extension matches the namespace of the service.
+              if (strpos(get_class($service), $config['screenshotExtension']) !== false) {
+                $imageUploaderService = $service;
```

Comment:
This code leads to the last of all matching services being selected as `imageUploaderService`, not the first. I assume this is intended behaviour?
If so, a short comment would be nice.

Reply:
That was not intentional. I added a break.
If you think we should change this behaviour, please let me know.